### ISSUE TITLE
[Bug]? Having to Sync twice for the orange dot to go away

### ISSUE THREAD

**Issue by tndk323** [ankidroid/Anki-Android#13528]
###### Reproduction Steps

1. I would press the Sync button and it would say "Collection Synchronized".
2. The Sync button remains orange despite syncing and I would have to press Sync again.
3. Says "Collection Synchronized" again and this time the orange dot finally goes away.

###### Expected Result
Isn't it supposed to Sync at once? I just wanted to know if it was normal to have to press Sync two times on AnkiDroid.

###### Actual Result
Having to press Sync twice.

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by MudssirAhmed** [comment]
@tndk323 Can you attach some recordings?

**Comment by tndk323** [comment]
https://user-images.githubusercontent.com/126623256/233693037-ff4d76f3-caa5-4c94-8913-c2ca1a18a3e0.mp4

**Comment by tndk323** [comment]
You can see that I have to click Sync twice for the orange dot to go away in the video.

**Comment by tndk323** [comment]
> @tndk323 Can you attach some recordings?

https://user-images.githubusercontent.com/126623256/233693313-b66739d0-41d9-4ea7-86e2-e11bd84669fd.mp4

You can see that I have to sync twice for the orange dot to go away.

**Comment by Arthur-Milchior** [comment]
@oakkitten I recall you mentioned that the top bar was updated too often. Do you know if it was corrected? Because, if so, I would expect it’s related, caused by the fact that the top bar is not updated anymore here

**Comment by khonkhortisan** [comment]
I sync and then open the ≡ hamburger menu for the dot to go away, does that work for you as well?

**Comment by oakkitten** [comment]
> Do you know if it was corrected?

Not, belike. I lack knowledge of such action

**Comment by tndk323** [comment]
> I sync and then open the ≡ hamburger menu for the dot to go away, does that work for you as well?

Yes, that does seem to work! Thank you. It's just that I know for a fact that it used to just sync at once when I had the older version of AnkiDroid straight from the Google Playstore, but this syncing issue only began after I downloaded updates from the GitHub. Maybe it's my phone's problem? Does nobody else have this issue?

**Comment by khonkhortisan** [comment]
ankidroid 2.16alpha98, Cricket Ovation 2, Android 11, Android System Webview 112.0.5615.136, notifications enabled, multimedia allowed, camera/mic allowed while in use, database allowed, quit permissions if app is not in use, app size 108 MB, data 3.36 GB, cache 30.20 MB, total 3.50 GB, data usage last month 102 MB, foreground 19.45 MB, background 82.73 MB, background allowed, without restriction disallowed, not battery optimized, always look for media to sync, don't sync automatically, show sync state, allow sync on metered connections, no custom sync server

**Comment by KendallPark** [comment]
I am also experiencing this issue and have been since downloading the alpha.

```
AnkiDroid Version = 2.16.0beta2

Android Version = 13

ProductFlavor = play

Manufacturer = Google

Model = Pixel 6a

Hardware = bluejay

Webview User Agent = Mozilla/5.0 (Linux; Android 13; Pixel 6a Build/TQ2A.230505.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/113.0.5672.132 Mobile Safari/537.36

ACRA UUID = c1fdac4b-f023-4fac-8832-56e4f6087c7d

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

**Comment by KendallPark** [comment]
```kotlin
            SyncStatus.markSyncCompleted()
            invalidateOptionsMenu()
            updateDeckList()
            WidgetStatus.update(this@createSyncListener)
```
In https://github.com/ankidroid/Anki-Android/blob/6c5c663917ef9f92ccb3d8dd83796a86a9a2ee6f/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt#L760

Does this need a call to `updateMenuState()`? 

```kotlin
            SyncStatus.markSyncCompleted()
            invalidateOptionsMenu()
            updateMenuState()
            updateDeckList()
            WidgetStatus.update(this@createSyncListener)
```

I did a bunch of digging and the only relevant place `SyncStatus.getSyncStatus` seems to get called is in `fetchSyncStatus` which is subsequently only called in `updateMenuState`.

**Comment by KendallPark** [comment]
Actually, I just figured out that `invalidateOptionsMenu()` should be calling `updateMenuState()` via `onCreateOptionsMenu()`. 

Does this `launchCatchingTask` start something async? 

```kotlin
        createMenuJob = launchCatchingTask {
            updateMenuState()
            updateMenuFromState(menu)
        }
```

https://github.com/KendallPark/Anki-Android/blob/bugfix-sync-badge-update/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt#L594

**Comment by khonkhortisan** [comment]
I enabled the experimental backend&v3 and now it syncs super-fast and the dot goes away. Is this bug secretly already fixed?

**Comment by mikehardy** [comment]
There are a few things secretly already fixed (and also super fast) about the new backend - we're hoping to make it the default pretty soon, ideally it will be the main feature in 2.17 - to make it default.

The orange dot should still go away after one sync on the legacy backend though...

**Comment by wujek-srujek** [comment]
I'm on 2.16.2 and I'm also experiencing this. Both workarounds (syncing twice; syncing once and opening the menu using the hamburger) work for me.

**Comment by miwucs** [comment]
I'm also seeing this with 2.16.2 (from the Play store). Clearly the sync does complete successfully, it's just the top bar UI not getting updated.

**Comment by david-allison** [comment]
Fixed in #14234 (2.16.3)

**Comment by wujek-srujek** [comment]
Fixed indeed, just tested with the new version. Thank you.
